### ISSUE TITLE
stripe.customers.createSource readme example returns not supported source error

### ISSUE THREAD

**Issue by dnuske** [stripe/stripe-node#369]
in the readme there is the following example:

```node
var stripe = require('stripe')('sk_test_....')
stripe.customers.create({
  email: 'foo-customer@example.com'
}).then(function(customer){
  return stripe.customers.createSource(customer.id, {
    source: {
       object: 'card',
       exp_month: 10,
       exp_year: 2018,
       number: '4242 4242 4242 4242',
       cvc: 100
    }
  });
})
```
When running it, returns the following error:
`{ Error: 'card[object Object]source[exp_month]=10[object Object]source[exp_year]=2018[object Object]source[number]=4242 4242 4242 4242[object Object]source[cvc]=100' is not a supported source object.`

**Comment by remi-stripe** [comment]
@dnuske I tried that exact example on my end with the latest node library and it just works as expected. 

Can you confirm which version of Node.js and stripe-node you're using? Can you also give us a bit more details about how you're running this code? It looks like there's some issue with the data you're sending that triggers the `source` parameter to not be sent as a hash potentially?

**Comment by ob-stripe** [comment]
Going to close this issue since we were unable to reproduce the problem.

@dnuske, if you're still observing this problem, feel free to reply here or directly reach out to Stripe's support at https://support.stripe.com/email.

**Comment by ORESoftware** [comment]
question - I see examples online where `source: 'tok_visa'`, what exactly does that mean? How can you create a new source with just a single string?

**Comment by ob-stripe** [comment]
@ORESoftware Normally, creating a charge or adding a payment source to a customer is a two-step process:

1. Client-side, in your frontend code, you collect and tokenize the customer's payment information using [Stripe.js](https://stripe.com/docs/stripe.js) or [Checkout](https://stripe.com/docs/checkout). This creates a "token" that you then send to your backend server.

2. Server-side, in your backend code, you use the token in an API request, e.g. to [create a charge](https://stripe.com/docs/api#create_charge) or [a customer](https://stripe.com/docs/api#create_customer).

This process ensures that the [PCI-sensitive](https://stripe.com/docs/security#validating-pci-compliance) card information never hits your server. Instead, your server only deals with tokens. This greatly simplifies the burden of PCI compliance.

When writing tests for your backend code, it would be cumbersome to have to simulate the first part of the process, so in test mode we provide "synthetic" tokens such as `tok_visa` that lets you simulate specific scenarios. You can find a complete list of such tokens here: https://stripe.com/docs/testing#cards.

Hope this helps! If you have more questions, please reach out to Stripe's support at https://support.stripe.com/email or join us on IRC in the #stripe channel on Freenode.

**Comment by gmeghalg** [comment]
@ob-stripe - Appreciate your help.

One more question...

1.) In my case, I'm using the "token" approach while attaching a source of customer.
await stripe.customers.createSource(customer.id, {source: token}); //token came from UI and customer I retrieved from .retrieve call

Now, how can
1.) Retrieve a source with 'token' for given customer id
2.) Update a source with 'token' for given customer id
3.) Delete a source with 'token' for given customer id

**Comment by remi-stripe** [comment]
Tokens `tok_1234` are one-time use. They can not be re-used after you use them to attach a card to a customer. When you do, the source returned will have a unique id `card_12345`. That is the id you would store in your own database to be able to retrieve update or delete the card in the future.

Since this is not related to a bug in the library, you can always follow up with our support team directly here: https://support.stripe.com/email

**Comment by Casen** [comment]
@remi-stripe after a lot of headaches and desire to throw my computer through a window, I have found this thread. In node, I have used the method you outlined earlier: client posts card details retrieves token, token sent back to server, and server adds card to customer

```
stripe.customers.createSource(customer.id, {source: token}) // token is a string here
```

When I do this, I receive the following error: 
```
"The source hash must include an 'object' key indicating what type of source to create."
```

Obviously... source is a string.... a token.... I am not sending a hash in this instance (per your documentation). Is the documentation incorrect?
